### DISCUSSION TITLE
Is there a way to use zmodule for gcloud completions?

### DISCUSSION THREAD

**Discussion by kyounger** [zimfw/zimfw#475] [Q&A]
The `google-cloud-sdk` is canonical place to get the completions for the `gcloud` cli tool. Currently I have it installed with homebrew, and the completions are included as part of that package. The approach Google uses is very wonky compared to most other zsh completion sources. You can view the contents of this file here: https://github.com/twistedpair/google-cloud-sdk/blob/master/google-cloud-sdk/completion.zsh.inc 

The documentation says to source this file and that does work. I'm looking to understand how to create a zmodule for this, though. I have attempted to create a my own module, but I'm not super familiar with zmodule and could use some direction on how to make this work.

https://github.com/kyounger/zsh-gcloud

**Comment by ericbn** [comment]
Hi @kyounger. Yeah, that Zsh completion implementation does not seem to be following the recommended/standard approach in some parts.

It does not seem intended to be autoloaded by compinit, so saving the file as a _gcloud file in a directory in the fpath might not have the desired effect.

I assumed adding
```
zmodule /usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk --source path.zsh.inc --source completion.zsh.inc
```
at the end of your ~/.zimrc file would work, but it didn't.

Doing
```
source /usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/completion.zsh.inc
```
in the command line in a new shell worked. Not sure what's wrong here.  :- (

**Comment by kyounger** [comment]
I really like zimfw, and moved to it when the whole zinit debacle occurred. I don't really miss it that much, other than this one thing. I'm not exactly sure how it worked, but the following syntax did something with this gcloud completion I can't seem to replicate.

```
zinit ice blockf
zinit snippet $MY_BREW_PREFIX/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/completion.zsh.inc
```

Not super helpful, but [this is the description](https://github.com/zdharma-continuum/zinit#completions) of that `blockf` approach:

> Disallow plugin to modify fpath. Useful when a plugin wants to provide completions in traditional way. Zinit can manage completions and plugin can be blocked from exposing them.

Not sure if this clears things up or further complicates them :)

**Reply by lnicola** [reply]
Off-topic, but what zinit debacle? :-)

**Comment by kyounger** [comment]
> Off-topic, but what zinit debacle? :-)

"debacle" might be a strong word. The original creator of zinit/zplug disappeared, removed the project, brought it back online — more than once. It was very strange and didn't endear many people to stick around. However it was forked since it was OSS and some awesome folks are now maintaining it that I would deem trustworthy. But during that whole situation I migrated away from it and found this project, which has been a much simpler interface for my zsh config. Zinit does have some interesting features, though.

**Reply by lnicola** [reply]
Got it, thanks. I wasn't aware of it.
